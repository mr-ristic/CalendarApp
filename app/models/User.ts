import { Instance, SnapshotIn, SnapshotOut, types } from 'mobx-state-tree';
import { withSetPropAction } from './helpers/withSetPropAction';

/**
 * Model description here for TypeScript hints.
 */
export const UserModel = types
  .model('User')
  .props({
    id: types.identifierNumber,
    firstName: types.string,
    lastName: types.string,
    email: types.string,
    avatar: types.string,
    timezone: types.string
  })
  .actions(withSetPropAction)
  .views((self) => ({})) // eslint-disable-line @typescript-eslint/no-unused-vars
  .actions((self) => ({})); // eslint-disable-line @typescript-eslint/no-unused-vars

export interface User extends Instance<typeof UserModel> {}
export interface UserSnapshotOut extends SnapshotOut<typeof UserModel> {}
export interface UserSnapshotIn extends SnapshotIn<typeof UserModel> {}
