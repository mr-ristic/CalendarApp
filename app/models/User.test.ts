import { getSnapshot, onSnapshot } from 'mobx-state-tree';
import { UserModel } from './User';
import mockUsers from '../../test/mockUsers';

describe('UserModel test', () => {
  it('should be created with correct props', () => {
    const oneUser = mockUsers.data[4];
    const user = UserModel.create(oneUser);

    expect(user.firstName).toEqual(mockUsers.data[4].firstName);
    expect(user.lastName).toEqual(mockUsers.data[4].lastName);
    expect(user.avatar).toEqual(mockUsers.data[4].avatar);
  });

  it('should create an instance of a model ', () => {
    const oneUser = mockUsers.data[6];
    const userInstance = UserModel.create(oneUser);

    expect(getSnapshot(userInstance)).toMatchSnapshot();
  });

  it('should update proprety with a action ', () => {
    const oneUser = mockUsers.data[9];
    const userInstance = UserModel.create(oneUser);

    let snapshot: any;

    onSnapshot(userInstance, (newSnapshot) => {
      snapshot = newSnapshot;
    });

    userInstance.setProp('email', 'mr@markoristic.net');
    expect(snapshot.email).toBe('mr@markoristic.net');
  });
});
