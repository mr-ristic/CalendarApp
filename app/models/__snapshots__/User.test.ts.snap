// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserModel test should create an instance of a model  1`] = `
{
  "avatar": "https://robohash.org/dictaaliquamquidem.png?size=50x50&set=set1",
  "email": "rloddon6@w3.org",
  "firstName": "Rita",
  "id": 7,
  "lastName": "Loddon",
  "timezone": "Europe/Warsaw",
}
`;
